vectordraw/public/js/vectordraw.js

```diff
@@ -0,0 +1,633 @@
+        if (this.drawMode) {
+            var coords = this.getMouseCoords(evt);
+            this.dragged_vector.point2.moveTo(coords.usrCoords);
+        }
+        if (this.dragged_vector) {
+            this.updateVectorProperties(this.dragged_vector);
+        }
+    };
+
+    VectorDraw.prototype.onBoardUp = function(evt) {
+        this.drawMode = false;
+        if (this.dragged_vector && !this.isVectorTailDraggable(this.dragged_vector)) {
+            this.dragged_vector.point1.setProperty({fixed: true});
+        }
+        this.dragged_vector = null;
+    };
+
+    VectorDraw.prototype.onEditStart = function(evt) {
+        var vectorName = $(evt.currentTarget).find('option:selected').data('vector-name');
+        var vectorObject = this.board.elementsByName[vectorName];
+        this.updateVectorProperties(vectorObject);
+    };
+
+    VectorDraw.prototype.onEditSubmit = function(evt) {
+        // Get vector that is currently "selected"
+        var vectorName = $('.element-list-edit', element).find('option:selected').data('vector-name');
+        // Get values from input fields
+        var newTail = $('.vector-prop-tail input', element).val(),
+            newLength = $('.vector-prop-length input', element).val(),
+            newAngle = $('.vector-prop-angle input', element).val();
+        // Process values
+        newTail = _.map(newTail.split(/ *, */), function(coord) {
+            return parseFloat(coord);
+        });
+        newLength = parseFloat(newLength);
+        newAngle = parseFloat(newAngle);
+        var values = [newTail[0], newTail[1], newLength, newAngle];
+        // Validate values
+        if (!_.some(values, Number.isNaN)) {
+            $('.vector-prop-update .update-error', element).hide();
+            // Use coordinates of new tail, new length, new angle to calculate new position of tip
+            var radians = newAngle * Math.PI / 180;
+            var newTip = [
+                newTail[0] + Math.cos(radians) * newLength,
+                newTail[1] + Math.sin(radians) * newLength
+            ];
+            // Update position of vector
+            var board_object = this.board.elementsByName[vectorName];
+            board_object.point1.setPosition(JXG.COORDS_BY_USER, newTail);
+            board_object.point2.setPosition(JXG.COORDS_BY_USER, newTip);
+            this.board.update();
+        } else {
+            $('.vector-prop-update .update-error', element).show();
+        }
+    };
+
+    VectorDraw.prototype.getVectorCoords = function(name) {
+        var object = this.board.elementsByName[name];
+        if (object) {
+            return {
+                tail: [object.point1.X(), object.point1.Y()],
+                tip: [object.point2.X(), object.point2.Y()]
+            };
+        }
+    };
+
+    VectorDraw.prototype.getState = function() {
+        var vectors = {}, points = {};
+        this.settings.vectors.forEach(function(vec) {
```

Comment:
You're calling the `.forEach()` method directly, which is not fully supported in Internet Explorer 9 and below. ([see caniuse](http://caniuse.com/#search=foreach)) However, Open edX only [supports Internet Explorer 10 and above](http://edx.readthedocs.org/projects/edx-installing-configuring-and-running/en/named-release-cypress/front_matter/browsers.html), so that's fine.

My question is, why do you call `.forEach()` directly here, and use `_.forEach()` in other places? The version in the Underscore library is just a proxy to the native version when the native version is defined, so there shouldn't be any performance difference. It's not a big deal, but the lack of consistency is a bit confusing, and I wanted to see if there was a reason that I don't know about.